static/js/tools/shared/tile_metadata_modal.tsx

```diff
@@ -0,0 +1,138 @@
+/**
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Displays a modal with links for each stat var to their page in Stat Var Explorer.
+ * TODO(beets): Don't update the modal with stat var names if it's already opened.
+ */
+
+import React, { useEffect, useState } from "react";
+import { Button, Modal, ModalBody, ModalFooter, ModalHeader } from "reactstrap";
+
+import { StatVarSpec } from "../../shared/types";
+import { datacommonsClient } from "../../utils/datacommons_client";
+
+const SV_EXPLORER_REDIRECT_PREFIX = "/tools/statvar#sv=";
+
+interface TileMetadataModalPropType {
+  statVarSpecs: StatVarSpec[];
+}
+
+type Dcid = string;
+type Name = string;
+type DcidNameTuple = [Dcid, Name];
+
+function buildMetadataRow(dcid: Dcid, name: Name, key: number): JSX.Element {
```

Comment:
Why not make this a proper component `MetadataRow` and refer to it like `<MetadataRow dcid={...} name={...} />` below? I find components easier to read since they have named parameters. you could also drop "key" as an explicit parameter if it's a component

Reply:
thanks - good point. updated.